src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallingGenerator.cs

```diff
@@ -20,6 +20,44 @@ public enum TargetFramework
         Net
     }
 
+    /// <summary>
+    /// An enumeration describing how a <see cref="TypePositionInfo"/> should be represented in its corresponding native signature element (parameter, field, or return value).
+    /// </summary>
+    public enum SignatureBehavior
+    {
+        /// <summary>
+        /// The native type should match the managed type, including rehydrating marshalling attributes and by-ref syntax (pure forwarding).
+        /// </summary>
+        ManagedTypeAndAttributes,
+        /// <summary>
+        /// The native signature should be the type returned by <see cref="IMarshallingGenerator.AsNativeType(TypePositionInfo)"/> passed by value.
+        /// </summary>
+        NativeType,
```

Comment:
Not going to press on this, just asking. If this is our future public API, do we want this to be `UnmanagedType`?

Reply:
If we change it, we should do a change through the whole `IMarshallingGenerator` API to change from `Native` to `Unmanaged`. I'm fine doing that, but I'd rather do that in a separate PR.